src/domain/opendatafabric/src/identity/dataset_identity.rs

```diff
@@ -223,6 +223,16 @@ impl DatasetName {
 
 ////////////////////////////////////////////////////////////////////////////////
 
+///////////////////////////////////////////////////////////////////////////////
+
+newtype_str!(
+    DatasetPattern,
+    Grammar::match_dataset_pattern,
+    DatasetPatternSerdeVisitor
+);
```

Comment:
`DatasetPattern` is a bad name as `Dataset` type is not some identifier - it's a complex aggregate object.

If this is a pattern that only matches dataset names we should name it `DatasetNamePattern`.